src/USER-PHONON/dynamical_matrix.cpp

```diff
@@ -0,0 +1,533 @@
+        conv_mass = 6.022E26; // kg -> g/mol
+        conv_distance = 1E-10; // meter -> angstrom
+
+    } else if (strcmp(style,"cgs") == 0) {
+        if (comm->me) error->warning(FLERR,"Conversion Warning: Multiplication by Large Float");
+        conv_energy = 6.022E12; // Erg -> 10 J/mol
+        conv_mass = 6.022E23; // g -> g/mol
+        conv_distance = 1E-7; // centimeter -> angstrom
+
+    } else if (strcmp(style,"electron") == 0) {
+        conv_energy = 262550; // Hartree -> 10 J/mol
+        conv_mass = 1; // amu -> g/mol
+        conv_distance = 0.529177249; // bohr -> angstrom
+
+    } else if (strcmp(style,"micro") == 0) {
+        if (comm->me) error->warning(FLERR,"Conversion Warning: Untested Conversion");
+        conv_energy = 6.022E10; // picogram-micrometer^2/microsecond^2 -> 10 J/mol
+        conv_mass = 6.022E11; // pg -> g/mol
+        conv_distance = 1E-4; // micrometer -> angstrom
+
+    } else if (strcmp(style,"nano") == 0) {
+        if (comm->me) error->warning(FLERR,"Conversion Warning: Untested Conversion");
+        conv_energy = 6.022E4; // attogram-nanometer^2/nanosecond^2 -> 10 J/mol
+        conv_mass = 6.022E5; // ag -> g/mol
+        conv_distance = 0.1; // angstrom -> angstrom
+
+    } else error->all(FLERR,"Units Type Conversion Not Found");
+
+}
+
+/* ---------------------------------------------------------------------- */
+
+void DynamicalMatrix::create_groupmap()
+{
+    //Create a group map which maps atom order onto group
+
+    int local_idx; // local index
+    int gid = 0; //group index
+    int nlocal = atom->nlocal;
+    int *mask = atom->mask;
+    bigint natoms = atom->natoms;
+    int *recv = new int[comm->nprocs];
+    int *displs = new int[comm->nprocs];
+    int *temp_groupmap = new int[natoms];
+
+    //find number of local atoms in the group (final_gid)
+    for (int i=1; i<=natoms; i++){
+        local_idx = atom->map(i);
+        if (mask[local_idx] & groupbit && local_idx < nlocal)
+            gid += 1; // gid at the end of loop is final_Gid
+    }
+    //create an array of length final_gid
+    int *sub_groupmap = new int[gid];
+
+    gid = 0;
+    //create a map between global atom id and group atom id for each proc
+    for (int i=1; i<=natoms; i++){
+        local_idx = atom->map(i);
+        if (mask[local_idx] & groupbit && local_idx < nlocal){
+            sub_groupmap[gid] = i;
+            gid += 1;
+        }
+    }
+
+    //populate arrays for Allgatherv
+    for (int i=0; i<comm->nprocs; i++){
+        recv[i] = 0;
+    }
+    recv[comm->me] = gid;
+    MPI_Allreduce(recv,displs,4,MPI_INT,MPI_SUM,world);
+    for (int i=0; i<comm->nprocs; i++){
+        recv[i]=displs[i];
+        if (i>0) displs[i] = displs[i-1]+recv[i-1];
+        else displs[i] = 0;
+    }
+
+    //combine subgroup maps into total temporary groupmap
+    MPI_Allgatherv(sub_groupmap,gid,MPI_INT,temp_groupmap,recv,displs,MPI_INT,world);
+    std::sort(temp_groupmap,temp_groupmap+group->count(igroup));
```

Comment:
You have to add `#include <algorithm>` if you want to use `std::sort()`. As an alternative, you could also use our internal c-style sort from the `mergesort.h` header.